src/rpc/services/SpectateService.ts

```diff
@@ -0,0 +1,531 @@
+                    const currentWindowDisputes =
+                        snapshotPayload.disputeWindows.shift()!;
+
+                    const creationTimestamp =
+                        await stateManager.stateChannelManagerContract.getDisputeWindowCreationTimestamp(
+                            channelId,
+                            currentForkId
+                        );
+
+                    // Reduce only this window
+                    const reducedOutput =
+                        await stateManager.stateChannelManagerContract.reduceProxyView(
+                            currentWindowDisputes,
+                            creationTimestamp
+                        );
+                    console.log(
+                        `Computed reduced output: forkGenesisTimestamp=${reducedOutput.forkGenesisTimestamp}`
+                    );
+
+                    // Move to next fork
+                    const nextForkId =
+                        reducedOutput.latestBlock.stateSnapshotHash;
+                    currentForkId = nextForkId;
+                    isDisputed =
+                        await stateManager.stateChannelManagerContract.isForkDisputed(
+                            channelId,
+                            currentForkId
+                        );
+                } else {
+                    // No disputes provided, can't proceed
+                    return {
+                        isValid: false,
+                        error: "No disputes provided for disputed fork"
+                    };
+                }
+            }
+
+            // Get the fork genesis snapshot from the participant's payload
+            const participantProvidedSnapshot =
+                snapshotPayload.latestForkGenesisSnapshot;
+
+            // Verify that the computed fork ID matches the participant's claimed fork ID
+            if (currentForkId !== participantProvidedSnapshot.forkId) {
+                return {
+                    isValid: false,
+                    error: `Computed fork ID ${currentForkId} does not match participant provided fork ID ${participantProvidedSnapshot.forkId}`
+                };
+            }
+
+            console.log(`Verified fork ID matches: ${currentForkId}`);
+
+            // Verify state proof to prove the latest state from the proven fork
+            if (
+                snapshotPayload.stateProof &&
+                snapshotPayload.stateProof.milestones.length > 0
+            ) {
+                console.log(`Verifying state proof from proven fork`);
+
+                // Get the latest state proof from the proven fork to compare with participant's state proof
+                const latestBlockHeight =
+                    stateManager.storage.blocks.getNextBlockHeight(
+                        currentForkId
+                    ) - 1;
+                const computedStateProof =
+                    await stateManager.agreementManager.getStateProof(
```

Comment:
why is the verifier calling this? Their storage is empty - the verification should be exlusivley a function of (the data provided and the data available on-chain)